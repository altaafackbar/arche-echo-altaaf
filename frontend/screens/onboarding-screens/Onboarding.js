import React from 'react';
import { Text, View, Button, StyleSheet, Image } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import Onboarding from 'react-native-onboarding-swiper';

const OnboardingScreen = () => {
    const navigation = useNavigation()
    return (
        <Onboarding
        onSkip={() => navigation.replace('MainMenu')}
        onDone={() => navigation.replace('MainMenu')}
        pages={[
            {
                backgroundColor: '#ffd44d',
                image: <Image source={require('../../assets/images/onboard-yellow.png')} style={styles.imageResizer} />,
                title: 'Mobile-Accessible Tools',
                subtitle: 'Access our resources from the palm of your hands. Anytime. Anywhere.',
            },
            {
                backgroundColor: '#76c1f9',
                image: <Image source={require('../../assets/images/onboarding-2.png')} style={styles.imageResizer} />,
                title: "Tools For Children's Health",
                subtitle: 'Learn about common childhood illnesses developed in collaboration with our partners and funders.',
            },
            
        ]} 
        />
    )

}

export default OnboardingScreen;

const styles = StyleSheet.create({
    container: {
        flex: 1, 
        alignItems: 'center', 
        justifyContent: 'center', 
    },
    imageResizer: {
        width: '100%', 
        height: '100%',
        resizeMode: 'center', 
        marginBottom: '-95%',
    },
})

// 76c1f9

// Test