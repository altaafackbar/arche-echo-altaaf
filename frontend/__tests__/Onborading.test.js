import React from "react";
import renderer from 'react-test-renderer';

import Onboarding from '../screens/onboarding-screens/Onboarding';


test('SignUp snapShot', () => {
    const snap = renderer.create(
        <Onboarding />
    ).toJSON();
    expect(snap).toMatchSnapshot();
});