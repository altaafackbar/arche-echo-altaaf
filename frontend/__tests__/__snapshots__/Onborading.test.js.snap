// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp snapShot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#ffd44d",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "backgroundColor": "#ffd44d",
          "image": <Image
            source={1}
            style={
              Object {
                "height": "100%",
                "marginBottom": "-95%",
                "resizeMode": "center",
                "width": "100%",
              }
            }
          />,
          "subtitle": "Access our resources from the palm of your hands. Anytime. Anywhere.",
          "title": "Mobile-Accessible Tools",
        },
        Object {
          "backgroundColor": "#76c1f9",
          "image": <Image
            source={1}
            style={
              Object {
                "height": "100%",
                "marginBottom": "-95%",
                "resizeMode": "center",
                "width": "100%",
              }
            }
          />,
          "subtitle": "Learn about common childhood illnesses developed in collaboration with our partners and funders.",
          "title": "Tools For Children's Health",
        },
      ]
    }
    disableVirtualization={false}
    extraData={null}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={1}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfig={
      Object {
        "itemVisiblePercentThreshold": 100,
      }
    }
    viewabilityConfigCallbackPairs={
      Array [
        Object {
          "onViewableItemsChanged": [Function],
          "viewabilityConfig": Object {
            "itemVisiblePercentThreshold": 100,
          },
        },
      ]
    }
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "paddingTop": 0,
              },
              null,
              Object {
                "height": 1334,
                "width": 750,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 0,
                  "paddingBottom": 60,
                  "width": "100%",
                },
                null,
              ]
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": "100%",
                  "marginBottom": "-95%",
                  "resizeMode": "center",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
              }
            }
          >
            <Text
              allowFontScaling={true}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 26,
                    "paddingBottom": 15,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#000",
                  },
                  Object {},
                ]
              }
            >
              Mobile-Accessible Tools
            </Text>
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
              }
            }
          >
            <Text
              allowFontScaling={true}
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, 0.7)",
                    "fontSize": 16,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.7)",
                  },
                  Object {},
                ]
              }
            >
              Access our resources from the palm of your hands. Anytime. Anywhere.
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "width": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.1)",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "paddingHorizontal": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexShrink": 1,
            "width": 200,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 0,
              "marginLeft": 10,
              "paddingHorizontal": 10,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 15,
                "left": 15,
                "right": 15,
                "top": 15,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 0,
                "opacity": 1,
              }
            }
          >
            <Text
              allowFontScaling={true}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.8)",
                  "fontSize": 16,
                }
              }
            >
              Skip
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.8)",
              "borderRadius": 3,
              "height": 6,
              "marginHorizontal": 3,
              "width": 6,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.3)",
              "borderRadius": 3,
              "height": 6,
              "marginHorizontal": 3,
              "width": 6,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexShrink": 1,
            "width": 200,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 0,
              "marginRight": 10,
              "paddingHorizontal": 10,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 15,
                "left": 15,
                "right": 15,
                "top": 15,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 0,
                "opacity": 1,
              }
            }
          >
            <Text
              allowFontScaling={true}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.8)",
                  "fontSize": 16,
                }
              }
            >
              Next
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
